six/common/builtins/aggregator.c

```diff
@@ -250,6 +251,10 @@ static PyObject *submit_event(PyObject *self, PyObject *args)
     // PyLong_AsLong will fail if called passing a NULL argument, be safe
     if (PyDict_GetItemString(event_dict, "timestamp") != NULL) {
         ev->ts = PyLong_AsLong(PyDict_GetItemString(event_dict, "timestamp"));
+        if (ev->ts == -1) {
```

Comment:
are you checking the return value because an `OverflowError` could be set? (https://docs.python.org/2/c-api/long.html#c.PyLong_AsLong)

If so, shouldn't we rather return NULL and bubble up the error instead of clearing it?

Not necessarily a blocker to merge this PR btw, simply handling the error is already an improvement

Reply:
This is on purpose to keep the old behavior. We would log something but the timestamp would get it's default value, 0.
 https://github.com/DataDog/datadog-agent/blob/fa373515349a7db22351afc8247399e45edac0b6/pkg/collector/py/api.go#L178-L186

Reply:
ok, makes sense, can you add an inline comment here to explain that we discard the error and set a `0` timestamp on purpose?